addons/product/tests/test_pricelist.py

```diff
@@ -198,3 +198,35 @@ def test_pricelists_res_partner_form(self):
 
         with Form(partner) as partner_form:
             self.assertEqual(partner_form.property_product_pricelist, self.sale_pricelist_id)
+
+    def test_pricelist_change_to_formula_and_back(self):
+        pricelist_1 = self.env['product.pricelist'].create({
+            'name': 'Sale pricelist',
+            'item_ids': [
+                Command.create({
+                    'compute_price': 'fixed',
+                    'fixed_price': 40,
+                    'base': 'list_price',  # based on public price
+                    'price_discount': 10,
+                    'applied_on': '3_global',
+                }),
+            ],
+        })
+        pricelist_2 = self.env['product.pricelist'].create({
+            'name': 'Sale pricelist',
+            'item_ids': [
+                Command.create({
+                    'compute_price': 'percentage',
+                    'percent_price': 20,
+                    'base': 'pricelist',  # based on pricelist_1
+                    'base_pricelist_id': pricelist_1.id,
+                    'applied_on': '3_global',
+                }),
+            ],
+        })
+        pricelist_item_form = Form(pricelist_2.item_ids[0])
+        pricelist_item_form.compute_price = 'formula'
+        pricelist_item_form.compute_price = 'percentage'
+        pricelist_item_form.percent_price = 20
+        pricelist_item_form.save()
```

Comment:
Using `with Form()` here makes the flow getting tested a bit clearer.
```suggestion
        with Form(pricelist_2.item_ids) as item_form:
            item_form.compute_price = 'formula'
            item_form.compute_price = 'percentage'
            item_form.percent_price = 20
```